core/src/main/scala/quasar/datasource/kafka/KafkaConsumerBuilder.scala

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.datasource.kafka
+
+import slamdata.Predef._
+
+import cats.effect.{ConcurrentEffect, ContextShift, Resource, Timer}
+import fs2.{Chunk, Stream}
+import fs2.kafka.{AutoOffsetReset, CommittableConsumerRecord, ConsumerSettings}
+import quasar.connector.MonadResourceErr
+
+class KafkaConsumerBuilder[F[_] : ConcurrentEffect : ContextShift : Timer : MonadResourceErr](
+    config: Config,
+    decoder: Decoder)
+    extends ConsumerBuilder[F] {
+
+  def mkConsumer: Resource[F, Consumer[F]] = {
+    val F: ConcurrentEffect[F] = ConcurrentEffect[F]
+    Resource.liftF(F.delay(f"${config.groupId}%s_${randomUuid().toString}%s")) map { groupId =>
+      // TODO: either get this to fail compilation if non-exhaustive, or move it somewhere it can return a configuration error
+      val recordDecoder = decoder match {
+        case Decoder.RawKey => KafkaConsumerBuilder.RawKey[F]
+        case Decoder.RawValue => KafkaConsumerBuilder.RawValue[F]
+      }
+
+      val consumerSettings = ConsumerSettings[F, Array[Byte], Array[Byte]]
+        .withAutoOffsetReset(AutoOffsetReset.Earliest)
+        .withBootstrapServers(config.bootstrapServers.toList.mkString(","))
+        .withGroupId(groupId)
+      //          .withBlocker(Blocker.liftExecutionContext(ec))
+
+      KafkaConsumer(consumerSettings, recordDecoder)
+    }
+  }
+}
+
+object KafkaConsumerBuilder {
+
+  def apply[F[_] : ConcurrentEffect : ContextShift : Timer : MonadResourceErr](
+      config: Config,
+      decoder: Decoder)
+      : ConsumerBuilder[F] =
+    new KafkaConsumerBuilder(config, decoder)
+
+
+  def RawKey[F[_]]: RecordDecoder[F, Array[Byte], Array[Byte]] =
+    (record: CommittableConsumerRecord[F, Array[Byte], Array[Byte]]) => Stream.chunk(Chunk.bytes(record.record.key))
+
+  def RawValue[F[_]]: RecordDecoder[F, Array[Byte], Array[Byte]] =
+    (record: CommittableConsumerRecord[F, Array[Byte], Array[Byte]]) => Stream.chunk(Chunk.bytes(record.record.value))
```

Comment:
What's the difference between choosing `RawKey` or `RawValue`? How does this affect how the data is decoded? Maybe an example with concrete data would help. Is this definitely an either/or, from the user's perspective, or might the user not want both?

Reply:
Just found the unit test that shows how we drop the key or drop the value. Could we not keep both, using the key as the field of a json object, and the value as the value? But perhaps this is in conflict with how people use the key/value in a kafka topic, and so wouldn't be what is expected?

Reply:
Each record has a key and a value. This datasource supports returning one or the other. I can certainly think of cases where the user might want both, but to do so the datasource needs to deserialize the data, and that's out of scope for the first version. These are "raw" decoders because they don't deserialize the data.

Reply:
👍 seems reasonable to include both key and value in a future version, if/when we need to support specific use cases that require this.